.github/workflows/backend.yml

```diff
@@ -0,0 +1,47 @@
+# This is a basic workflow to help deploy the backend code to Azure server
+# it triggers on push and pull request events to the master branch
+# it runs the jobs on the API/ directory and deploys using secret credentials
+# Update the secret credentials with details from the Azure web server.
+
+name: "CI/CD Pipeline for Azure webapp deployment"
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened, closed]
+    branches:
+      - master
+jobs:
+  build_and_deploy:
+    name: Build test and Deploy Node App to Azure
+    runs-on: ubuntu-latest
+    if: >-
+      github.event_name == 'push' || (github.event_name == 'pull_request' &&
+      github.event.action != 'closed')
+    environment: production
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: npm install, build, and test
+        run: |
+          pwd
+          ls
+          cd API/
+          npm install
+          npm run build --if-present
+          npm run test --if-present
+          echo Successful
+      - name: "Deploy to Azure"
+        uses: azure/webapps-deploy@v2
+        with:
+          app-name: ${{ secrets.AZURE_WEBAPP_NAME }}
+          publish-profile: ${{ secrets.AZURE_WEBAPP_PUBLISH_PROFILE }}
+          package: ${{ secrets.AZURE_WEBAPP_PACKAGE_PATH }}
+      - if: github.ref == 'refs/heads/master' && job.status == 'success'
+        run: |
+          git push -u origin HEAD:master -f
```

Comment:
this is interesting. it deploys in the previous step, and if successful auto-merges the code. but i don't think I like that approach. I'd rather `only` deploy if it is master. so maybe better to move this `if` statement to the previous deploy step, and so that we only deploy after the PR is closed and merged to master